examples/jsm/loaders/MTLLoader.js

```diff
@@ -365,6 +365,8 @@ class MaterialCreator {
 
 			}
 
+			map.sourceFile = value; // for UITexture
```

Comment:
No loader sets `sourceFile`. It is not even a property of `Texture`. 

I understand `sourceFile` is used in context of the editor but addons should not implement any sort of editor specific logic. So I don't think this change is acceptable.

Reply:
Can you please explain why changes to `MTLLoader` are necessary in the first place?

Reply:
OBJLoader takes MTLLoader artifacts as `materials` to generate objects, and those materials maps (if any) do not bind filename to anywhere, so the cause belong to MTLLoader. :)